sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -255,3 +262,121 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+case class Round(left: Expression, right: Expression)
+  extends Expression with trees.BinaryNode[Expression] with Serializable {
+
+  def this(left: Expression) = {
+    this(left, Literal.create(0, IntegerType))
+  }
+
+  override def nullable: Boolean = left.nullable || right.nullable
```

Comment:
This may not hold if we would behave similarly with Hive, NaN and Infinity would result in null.